### PR TITLE
Updating the actions to push microservice images to public repos.

### PR DESCRIPTION
**Pull request by scottwn**
This commit also contains lots of style changes due to `pre-commit run
--all-files`.

I can split those changes into another PR if it's too difficult to merge
them now.

AgnostiqHQ/covalent-infra#18

- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

### CODE REVIEW

**Review by Emmanuel289 — COMMENTED** [review]
Should we assign DATA_SVC_PORT to a different number than 8004 since the UI service is using both 8004 and 8005?

**Comment by scottwn** [comment]
In the cloud environment the port doesn't matter.... maybe we should assign them all to `80`?

**Comment by Emmanuel289** [comment]
> In the cloud environment the port doesn't matter.... maybe we should assign them all to `80`?

Okay, that works.

File: .github/workflows/publish_develop.yml [hunk 1]
```diff
@@ -25,8 +25,8 @@ on:
     branches:
       - develop
     paths-ignore:
-      - 'CHANGELOG.md'
-      - 'VERSION'
+      - "CHANGELOG.md"
```
Comment (wjcunningham7) [inline .github/workflows/publish_develop.yml:28]: does single vs double quote matter in the actions?

Comment (scottwn) [reply .github/workflows/publish_develop.yml:28]: It does if you're doing variable expansion or using some related shell commands, but it doesn't matter here. This is enforcing the precommit rule.

**Comment by scottwn** [comment]
@Emmanuel289 @venkatBala I'm thinking more about the right way to do the port mapping in the docker compose files. I was thinking that we can assign everything to the same port in `docker-compose.yml` since each service will be running on a separate address. But I guess that would break it if you wanted to run it locally? But then again... maybe the purpose of `docker-compose.yml` isn't for local hosting? What do you think?

File: .github/workflows/publish_develop.yml [hunk 2]
```diff
@@ -69,3 +69,20 @@ jobs:
...
+          docker tag covalent-queuer:latest ${{ secrets.QUEUER_DEV_REPO }}
+          docker tag covalent-results:latest ${{ secrets.RESULTS_DEV_REPO }}
+          docker tag covalent-runner:latest ${{ secrets.RUNNER_DEV_REPO }}
+          docker tag covalent-ui:latest ${{ secrets.UI_DEV_REPO }}
+          docker push ${{ secrets.DATA_DEV_REPO }}
+          docker push ${{ secrets.DISPATCHER_DEV_REPO }}
+          docker push ${{ secrets.QUEUER_DEV_REPO }}
+          docker push ${{ secrets.RESULTS_DEV_REPO }}
+          docker push ${{ secrets.RUNNER_DEV_REPO }}
+          docker push ${{ secrets.UI_DEV_REPO }}
```
Comment (venkatBala) [inline .github/workflows/publish_develop.yml:88]: should we use secrets for the public ECR registries?

Comment (scottwn) [reply .github/workflows/publish_develop.yml:88]: That's the standard that we've been using. Obviously the repos themselves are not secret, but we have put all the ECR repos into secrets for consistency.

Comment (venkatBala) [reply .github/workflows/publish_develop.yml:88]: sounds good!

**Comment by scottwn** [comment]
I guess we can't assign both the backend and the frontend on the UI to port `80`.... how should we map the UI ports?

**Comment by Emmanuel289** [comment]
> @Emmanuel289 @venkatBala I'm thinking more about the right way to do the port mapping in the docker compose files. I was thinking that we can assign everything to the same port in `docker-compose.yml` since each service will be running on a separate address. But I guess that would break it if you wanted to run it locally? But then again... maybe the purpose of `docker-compose.yml` isn't for local hosting? What do you think?

I was thinking about this point as well. `docker-compose.yml` needs unique port allocations for local hosting. Maybe supervisord is a better alternative for running the services locally

**Comment by scottwn** [comment]
yeah I think `docker-compose.yml` should not be designed to run locally

**Comment by Emmanuel289** [comment]
> I guess we can't assign both the backend and the frontend on the UI to port `80`.... how should we map the UI ports?

80 and 8080?

**Comment by scottwn** [comment]
which should be which? `80:8004` and `8000:8005`?

**Comment by Emmanuel289** [comment]
> which should be which? `80:8004` and `8080:8005`?

Yes, these should be good

**Comment by scottwn** [comment]
We removed `8005` because it seemed like it wasn't doing anything. Was that a mistake?

File: docker-compose.yml [hunk 3]
```diff
@@ -6,62 +6,58 @@ version: "3.9"
...
   covalent-ui:
-    image: "${UI_REPO}"
+    image: "public.ecr.aws/covalent/covalent_ui"
     ports:
-      - 3001:8004
-      - 8005:8005
-    environment:
-      UI_SVC_HOST: 0.0.0.0
-      UI_SVC_PORT: 8004
+      - "80:8004"
```
Comment (scottwn) [inline docker-compose.yml:60]: @Emmanuel289 can you confirm that this piece looks OK from your side?

Comment (Emmanuel289) [reply docker-compose.yml:60]: yeah, this looks good

**Pull request merged**